roles/kubernetes/master/templates/kube-proxy-ds.yaml.j2

```diff
@@ -0,0 +1,77 @@
+apiVersion: extensions/v1beta1
+kind: DaemonSet
+metadata:
+  name: kube-proxy
+  labels:
+    k8s-app: kube-proxy
+spec:
+  template:
+    metadata:
+      annotations:
+        scheduler.alpha.kubernetes.io/critical-pod: ""
+      creationTimestamp: null
+      labels:
+        k8s-app: kube-proxy
+    spec:
+      containers:
+      - command:
+        - /usr/local/bin/kube-proxy
+        - --config=/var/lib/kube-proxy/config.conf
+        - --hostname-override=$(NODE_NAME)
+        env:
+        - name: NODE_NAME
+          valueFrom:
+            fieldRef:
+              apiVersion: v1
+              fieldPath: spec.nodeName
+        image: gcr.io/google-containers/kube-proxy:{{ kube_version }}
+        imagePullPolicy: IfNotPresent
+        name: kube-proxy
+        resources: {}
+        securityContext:
+          privileged: true
+          procMount: Default
+        terminationMessagePath: /dev/termination-log
+        terminationMessagePolicy: File
+        volumeMounts:
+        - mountPath: /var/lib/kube-proxy
+          name: kube-proxy
+        - mountPath: /run/xtables.lock
+          name: xtables-lock
+        - mountPath: /lib/modules
+          name: lib-modules
+          readOnly: true
+      dnsPolicy: ClusterFirst
+      hostNetwork: true
+      nodeSelector:
+        beta.kubernetes.io/os: linux
```

Comment:
So the node selector patch can be removed - https://github.com/kubernetes-sigs/kubespray/blob/master/roles/win_nodes/kubernetes_patch/tasks/main.yml#L10

In fact, the whole win_nodes role could probably be removed cause all it does is the node selector patch.

Reply:
`nodeSelector` can be removed

Reply:
Some individuals use kubespray to provision the linux side of a mixed OS cluster.  The node selector is there to prevent the kube-proxy ds from running on windows nodes.